core/src/identity.rs

```diff
@@ -63,7 +63,7 @@ use std::convert::{TryFrom, TryInto};
 /// let keypair = Keypair::rsa_from_pkcs8(&mut bytes);
 /// ```
 ///
-#[derive(Clone)]
+#[derive(Debug, Clone)]
```

Comment:
I would like to be extra careful with private key material. E.g. I would like to prevent users ever accidentally printing private key material to stdout, potentially being persisted in some log store, ...

How about a custom `Debug` implementation that only prints the key type. In case specific `xxx::Keypair` `Debug` implementations ever accidentally leak private key material, it would still not be exposed.

Next to the key type we could include the `PeerId`.

Reply:
Would one not get the same benefit from the `PeerId`?